protocols/gossipsub/src/layer.rs

```diff
@@ -929,6 +920,31 @@ impl<TSubstream> Gossipsub<TSubstream> {
 
         gossip_peers[..n].to_vec()
     }
+
+    // adds a control action to control_pool
+    fn control_pool_add(&mut self, peer: PeerId, control: GossipsubControlAction) {
```

Comment:
Thus function could be replaced like so:
```suggestion
    fn control_pool_add(&mut self, peer: PeerId, control: GossipsubControlAction) {
        (*self
            .control_pool
            .entry(peer.clone())
            .or_insert_with(|| Vec::new()))
        .push(control.clone());
    }
```

Reply:
Nice. I struggled to find a succinct way of doing this.